lib/langchain/agent/function_call_agent/function_call_agent.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Langchain::Agent
+  class FunctionCallAgent < Base
+    attr_reader :conversation, :tools
+
+    #
+    # Initializes the Agent
+    #
+    # @param llm [Object] The LLM client to use
+    # @param tools [Array<Object>] The tools that can be invoked
+    #
+    def initialize(llm:, tools:)
```

Comment:
should we validate here if the llm supports openai? if llm is not that particular one it probably will fail with an unexpected error right? maybe is better to be proactive here so its clear to the developer the error, and don't need to debug whats going on/why is failing

Reply:
another option is to not receive the llm but create it on the agent 🤔
I don't know about other models supporting function calling, but probably will not be an exact implementation, so if that's true, for now maybe is worth to make the implementation specific to openai api?

Reply:
FTR, this is the current behavior:
```ruby
irb(main):003:0> a = Langchain::Agent::FunctionCallAgent.new(llm: Langchain::LLM::AI21.new(api_key: '123'), tools: [Langchain::Tool::Calculator.new])
=>
#<Langchain::Agent::FunctionCallAgent:0x000000010362f008
...
irb(main):004:0> a.run(question: 'a')
/Users/user/dev/langchainrb/lib/langchain/conversation.rb:41:in `set_functions': undefined method `functions=' for #<Langchain::LLM::AI21:0x00000001040d2d20 @client=#<AI21::Client:0x00000001042d0ac8>, @defaults={:temperature=>0.0, :model=>"j2-ultra"}> (NoMethodError)```